nncf/experimental/torch/sparsity/movement/structured_mask_strategy.py

```diff
@@ -1,206 +0,0 @@
-from nncf.torch.nncf_network import NNCFNetwork
-
-STRUCTURED_MASK_STRATEGY = Registry('structured_mask_strategy')
-
-
-def detect_supported_model_family(model: NNCFNetwork) -> Optional[str]:
-    """
-    Returns the model family name if the model is supported by movement sparsity to conduct structured
-    masking. Such name can be used as the key in `STRUCTURED_MASK_STRATEGY` to get the corresponding
-    structured mask strategy.
-
-    :param model: The compressed model wrapped by NNCF.
-    :return: The string name of the model family that `model` belongs to. If `model` is not supported,
-        then returns None.
-    """
-    model_pymodules = inspect.getmodule(model).__name__.split('.')
-    if len(model_pymodules) >= 3 and model_pymodules[:2] == ['transformers', 'models']:
-        # the case of input model defined by HuggingFace's transformers
-        model_family = f'huggingface_{model_pymodules[2]}'
-        if model_family in STRUCTURED_MASK_STRATEGY.registry_dict:
-            return model_family
-    return None
-
-
-class StructuredMaskRule:
-    """
-    Defines the rule to resolve the structured mask in a sparsifiable layer.
-    """
-
-    def __init__(self,
-                 keywords: Union[List[str], str],
-                 prune_by_row: bool,
-                 prune_grid: Tuple[int, int]):
-        """
-        :param keywords: The patterns of module_node_name.
-        :param prune_by_row: Whether the matched module should be pruned by row or column.
-        :param prune_grid: The grid that should be regarded as a whole for structured mask resolution.
-        """
-        self.keywords: List[str] = [keywords] if isinstance(keywords, str) else keywords
-        self.prune_by_row = prune_by_row
-        self.prune_grid = prune_grid
-
-    def __str__(self) -> str:
-        return '{class_name}({attrs})'.format(
-            class_name=self.__class__.__name__,
-            attrs=', '.join(f'{k}={v}' for k, v in self.__dict__.items())
-        )
-
-
-class BaseStructuredMaskStrategy(ABC):
-    """
-    Strategy that defines how structured masking is conducted for certain
-    building blocks in a model.
-    """
-
-    @classmethod
-    @abstractmethod
-    def from_compressed_model(cls, compressed_model: NNCFNetwork):
-        """
-        Initializes the structured mask strategy from the compressed model.
-
-        :param compressed_model: The model wrapped by NNCF.
-        """
-
-    @property
-    @abstractmethod
-    def rules_by_group_type(self) -> Dict[BuildingBlockType, List[StructuredMaskRule]]:
-        """
-        Returns the rule list for each `BuildingBlockType`, which should cover all
-        modules that will do structured masking as a group.
-
-        :return: A dict specifying `StructuredMaskRule` list for each `BuildingBlockType`.
-        """
-
-
-class BaseTransformerStructuredMaskStrategy(BaseStructuredMaskStrategy, ABC):
```

Comment:
@vuiseng9 @yujiepan-work I'm going to delete this file with strategies, since we're moving to the automatic structured pruning